src/slots/InSlot.js

```diff
@@ -41,6 +41,13 @@ class InSlot extends Slot {
     return derivedUnit || (this.sourceSlot && this.sourceSlot.derivedUnit) || this.unit
   }
 
+  get InputNodeId(){
+      const inputNode = this.connectedFunctionalities
+        .find(({ type, source }) => (
+          type === 'InputNode' && source.name === 'ospin-webapp'))
```

Comment:
tried to referencene those but whenever i imported the file it broke

Reply:
let's change it to be lower case - this is not C# :P

Reply:
I think you can solve the issue of circular references by putting this functions on the graph itself. The function could take the fctId and the slotName as arguments, and then do 

```js
getInputNodeId(fctId, slotName) {
  const fct =  this.getFunctionalityByUd(fctId)
  const slot = fct.getSlotByName(slotName)
  ...
  return inputNode ? inputNode.id : null
}
```

but I see how it is breaking a little bit with the current pattern. We haven't thought about a cleaner separation of what should be put where, so for now it is fine.